Fix tooltip going out of bounds

dhruvparmar372 opened the pull request:
Fixes https://github.com/yjose/reactjs-popup/issues/4 by adding ability to provide an array of position enums which will be applied to the tooltip in the same order and picks the first position which doesn't go outside the bounds of element returned by `getTooltipBoundary` (default `window`)

Changes
1. Allow sending array via `position` prop
3. Add prop `getTooltipBoundary` 
4. Changes `testEnvironment` from `node` -> `jsdom` due to following error
```It looks like you called `mount()` without a global document being loaded.```

yjose commented on package.json:
```diff
@@ -2,11 +2,11 @@
-  "jsnext:main": "reactjs-popup.es.js",
-  "unpkg": "reactjs-popup.js",
-  "jsdelivr": "reactjs-popup.js",
+  "main": "lib/reactjs-popup.cjs.js",
```
here, you shouldn't add the lib directory in entry file  because we publish the lib directory to NPM registry, not the whole project

dhruvparmar372 commented:
@yjose I've added ability to pass down an `Array` via `position` prop, similar to `on` prop. Fixed package.json and added example for bounded toooltip

yjose commented:
Great work @dhruvparmar372
